### ISSUE TITLE
Software renderer not working

### ISSUE THREAD

**Issue by YakBizzarro** [tobiaspfaff/grim_mouse#12]
If I select the software renderer, this mod is not working correctly. There are two main issues:
- The cursor is surrounded by a white square (see image)
  ![cursor](https://f.cloud.github.com/assets/767740/2296075/2447e602-a099-11e3-85d6-377853e9a13a.jpeg)

-If I ask Manny to move by clicking on a free region (ie. not hotspots, normal pointer) the program stop with the error "hitting normal at 90 degrees!"

Debian Jessie (testing) x86-64
Git revision 585a46f

**Comment by tobiaspfaff** [comment]
Yeah, I currently rely on glu_Unproject to implement one of the Opcodes. I'll see how to do this stuff without GL...

**Comment by tobiaspfaff** [comment]
It looks like its not easily possible to add transparency in TinyGL (at least I couldn't figure out how), so it looks like the mod will remain GL only for now. 
Is there a good reason to support the software renderer?

**Comment by YakBizzarro** [comment]
I'm not an expert of TinyGL, maybe you can look at font rendering code, which it seems me that is doing some transparency stuff. Anyway, the TinyGL renderer is not perfect, there is a GSoC proposal to improve it.
TinyGL is useful when there are problems with the OpenGL acceleration or when is not available.

**Comment by tobiaspfaff** [comment]
Ah, good point, there has to be some sort of transparency handling there. I'll have a look once I have a bit more time.
